packages/calcite-components/src/components/block-group/block-group.tsx

```diff
@@ -287,18 +304,43 @@ export class BlockGroup extends LitElement implements InteractiveComponent, Sort
     const toEl = moveTo.element as BlockGroup["el"];
     const fromElItems = Array.from(fromEl.children).filter(isBlock);
     const oldIndex = fromElItems.indexOf(dragEl);
+    const newIndex = 0;
 
     if (!fromEl) {
       return;
     }
 
+    if (
+      fromEl.canPull?.({
+        toEl,
+        fromEl,
+        dragEl,
+        newIndex,
+        oldIndex,
+      }) === false
+    ) {
+      this.calciteBlockGroupPullFail.emit({ toEl, fromEl, dragEl, oldIndex, newIndex });
+      return;
+    }
+
+    if (
+      toEl.canPut?.({
+        toEl,
+        fromEl,
+        dragEl,
+        newIndex,
+        oldIndex,
+      }) === false
+    ) {
+      toEl.putFailed({ toEl, fromEl, dragEl, oldIndex, newIndex });
```

Comment:
It looks like `putFailed` can be inlined since it seems to only be used here. Applies to `list`.

Reply:
@jcfranco what do you mean by inlined? Can you elaborate pretty plz :)

Reply:
You can ignore this one. I thought `putFailed` wasn't accessed externally.